### ISSUE TITLE
include text input clear icon in tab order

### ISSUE THREAD

**Issue by mirrorstage** [CDCgov/prime-simplereport#6536]
## Area or Component

The "patient search" input field found on `Conduct tests`, on `Results`, and on `Patients`.

## Quick Summary

The patient search input fields present throughout the application use the typical pattern of presenting an `x` ornament/decoration once the user has entered at least 1 character. When the user clicks or taps the `x`, the field is cleared of all input. 

However, this interactive element is not included in the tab order. 

### How to Replicate

1. Using a device with a keyboard, navigate to the SimpleReport demo site, then navigate to the `Conduct tests` tab.
2. Enter a few characters of input in the `Search for a patient...` field
3. The `x` appears in the text field at the right
4. Press the tab key

#### Expected

* The application shifts focus to the `x`

#### Found

* On `Conduct tests`, focus shifts to the `Add new patient` button 
* On `Results`, focus shifts to the `Add new patient` button
* On `Patients`, at desktop width focus shifts to the `How can we improve?` CTA. At narrower widths, it goes to the browser URL field

## Accessibility Violation

Per the digital.gov accessibility guidance found [here](https://accessibility.digital.gov/ux/tab-order/), "user should be able to use the tab key to navigate to every interactive element on a page." This resource further references these WCAG guidelines:

* [1.3.2 Meaningful Sequence](https://www.w3.org/WAI/WCAG20/quickref/#content-structure-separation-sequence)
* [2.4.3 Focus Order](https://www.w3.org/WAI/WCAG20/quickref/?showtechniques=14%2C128&currentsidebar=%23col_overview#navigation-mechanisms-focus-order)
* [2.4.7 Focus Visible](https://www.w3.org/WAI/WCAG20/quickref/?showtechniques=14%2C128&currentsidebar=%23col_overview#navigation-mechanisms-focus-visible)

## Action Requested

For each text input component, add the `x` to the tab order

## Screenshots (optional)

Conduct tests

<img width="1508" alt="image" src="https://github.com/CDCgov/prime-simplereport/assets/6709726/46b84a70-992a-405a-9b45-11cd86d13012">

## Source Code (optional)

N/A

## Original Deque Issue

N/A

**Comment by BobanL** [comment]
Looked into this a little bit this morning. It seems this is clear text "x" button is something implemented by browsers. For example, Firefox does not have the "x". To resolve this issue we could either just [replace the type to be "text"](https://github.com/CDCgov/prime-simplereport/blob/2b24fc8ee0a7390f3b368208f69c3627e1c69a01/frontend/src/app/testQueue/addToQueue/SearchInput.tsx#L65) and set [aria-role to searchbox](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/searchbox_role) OR possibly we can remove it the [css](https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-search-cancel-button)

**Comment by mirrorstage** [comment]
Interesting -- you're right, I don't see this when I use FF. 

I would tend to think that the aria approach has the edge on changing the type, feels like it makes the intention more explicit in the code. The linked doc makes the CSS approach feel dodgy.
